src/Core/Query/Helper.php

```diff
@@ -113,52 +134,50 @@ public function escapePhrase(string $input): string
      *
      * @return string|bool false is returned in case of invalid input
      */
-    public function formatDate($input)
+    public function convertAndFormatDate($input)
     {
         switch (true) {
             // input of datetime object
             case $input instanceof \DateTimeInterface:
                 $input = clone $input;
+
                 break;
             // input of timestamp or date/time string
             case \is_string($input):
             case is_numeric($input):
                 // if date/time string: convert to timestamp first
-                if (\is_string($input)) {
-                    $input = strtotime($input);
+                if (\is_string($input) && false === $input = strtotime($input)) {
+                    return false;
                 }
 
                 // now try converting the timestamp to a datetime instance, on failure return false
                 try {
                     $input = new \DateTime('@'.$input);
                 } catch (\Exception $e) {
-                    $input = false;
+                    return false;
                 }
                 break;
-            // any other input formats can be added in additional cases here...
-            // case $input instanceof Zend_Date:
-
             // unsupported input format
             default:
-                $input = false;
-                break;
+                return false;
         }
 
-        // handle the filtered input
-        if ($input) {
-            // when we get here the input is always a datetime object
-            $input = $input->setTimezone(new \DateTimeZone('UTC'));
-            // Solr seems to require the format PHP erroneously declares as ISO8601.
-            /** @noinspection DateTimeConstantsUsageInspection */
-            $iso8601 = $input->format(\DateTimeInterface::ISO8601);
-            $iso8601 = strstr($iso8601, '+', true); //strip timezone
-            $iso8601 .= 'Z';
-
-            return $iso8601;
-        }
+        // when we get here the input is always a datetime object
+        $input = $input->setTimezone(new \DateTimeZone('UTC'));
+
+        return $this->dateToSolrUtcString($input);
+    }
 
-        // unsupported input
-        return false;
+    /**
+     * Format date time interface to date string required by Solr.
+     *
+     * @param \DateTimeInterface $dateTime
```

Comment:
I would add some more explanations here:

Be aware this method just formats the date time as required by Solr without converting it into the UTC TimeZone. If your PHP application works with different time zones and doesn't convert to UTC before calling this function this might lead to unexpected errors when doing "date math". Consider Helper::convertToUtcAndFormatDate() instead.

Reply:
yes, agreed. 

perhaps a little remark about the "out of sync" issue we've discussed before?
so basically one should use either method and not both depending on their workflow